vendor/github.com/rancher/rke/cluster/authn-webhook.go

```diff
@@ -0,0 +1,155 @@
+// 	CurrentContext string `yaml:"current-context"`
+// 	Contexts       []AuthnWebhookConfigContextInfo
+// }
+
+// func generateAuthnWebhookConfigYAML(clusterName string) string {
+// 	authnWebhookConfig, _ := yaml.Marshal(AuthnWebhookConfig{
+// 		ApiVersion: "v1",
+// 		Kind:       "Config",
+// 		Clusters: []AuthnWebhookConfigClusterInfo{
+// 			AuthnWebhookConfigClusterInfo{
+// 				Name: clusterName,
+// 				Cluster: AuthnWebhookConfigCluster{
+// 					InsecureSkipTLSVerify: true,
+// 					Server:                "https://d5469f2a.ngrok.io/authenticate",
+// 				},
+// 			},
+// 		},
+// 		Users: []AuthnWebhookConfigUserInfo{
+// 			AuthnWebhookConfigUserInfo{
+// 				Name: "authn-webhook-" + clusterName,
+// 				User: AuthnWebhookConfigUser{
+// 					InsecureSkipTLSVerify: true,
+// 				},
+// 			},
+// 		},
+// 		CurrentContext: "webhook",
+// 		Contexts: []AuthnWebhookConfigContextInfo{
+// 			AuthnWebhookConfigContextInfo{
+// 				Name: "webhook",
+// 				Context: AuthnWebhookConfigContext{
+// 					User:    "authn-webhook-" + clusterName,
+// 					Cluster: clusterName,
+// 				},
+// 			},
+// 		},
+// 	})
+// 	return string(authnWebhookConfig)
+// }
+
+func generateAuthnWebhookConfigYAML(clusterName string) string {
+	return `apiVersion: v1
+kind: Config
+clusters:
+- name: Default
+  cluster:
+    insecure-skip-tls-verify: true
+    server: http://127.0.0.1:6440/v1/authenticate
+users:
+- name: Default
+  user:
+    insecure-skip-tls-verify: true
+current-context: webhook
+contexts:
+- name: webhook
+  context:
+    user: Default
+    cluster: Default
+`
+}
+
+func (c *Cluster) DeployAuthnWebhookConfig(ctx context.Context, uniqueHosts []*hosts.Host, alpineImage string, prsMap map[string]v3.PrivateRegistry) error {
+	authnYAML := generateAuthnWebhookConfigYAML(c.ClusterName)
+	for _, host := range uniqueHosts {
+		log.Infof(ctx, "[%s] Deploying webhook config file to node [%s]:\n%s", AuthnWebhookConfigServiceName, host.Address, authnYAML)
+		if err := doDeployAuthnWebhookConfigFile(ctx, host, authnYAML, alpineImage, prsMap); err != nil {
+			return fmt.Errorf("Failed to deploy webhook config file on node [%s]: %v", host.Address, err)
+		}
+	}
+
+	return nil
+}
+
+func doDeployAuthnWebhookConfigFile(ctx context.Context, host *hosts.Host, yaml, alpineImage string, prsMap map[string]v3.PrivateRegistry) error {
+	// remove existing container. Only way it's still here is if previous deployment failed
+	if err := docker.DoRemoveContainer(ctx, host.DClient, AuthnWebhookConfigDeployer, host.Address); err != nil {
+		return err
+	}
+	imageCfg := &container.Config{
+		Image: alpineImage,
+		Cmd: []string{
+			"sh",
+			"-c",
+			fmt.Sprintf("t=$(mktemp); echo -e \"%s\" > $t && mv $t %s && chmod 644 %s", yaml, AuthnWebhookConfigFile, AuthnWebhookConfigFile),
+		},
+	}
+	hostCfg := &container.HostConfig{
+		Binds: []string{
+			fmt.Sprintf("%s:/etc/kubernetes:z", path.Join(host.PrefixPath, "/etc/kubernetes")),
+		},
+		Privileged: true,
```

Comment:
why does it need to be privileged?

Reply:
Fixed, it shouldn't need to be, probably same for cloud-provider.go